Fix fonts which are bigger variants of Font6x8

caemor opened the pull request:
Hi! Thank you for helping out with Embedded Graphics development! Please:

- [x] Check that you've added passing tests and documentation
- [x] Add a simulator example(s) where applicable
- [x] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) and appropriate crate (`embedded-graphics`, `simulator`, `tinytga`, `tinybmp`) if your changes affect the **public API**
- [x] Run `rustfmt` on the project
- [x] Run `./build.sh` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

Font12x16 and Font24x32 were using fallback in the charset for areas where it wasn't necessary.

Still need to run more tests

jamwaffles commented on embedded-graphics/CHANGELOG.md:
```diff
@@ -15,6 +15,7 @@ Embedded Graphics is a `no_std` library for adding graphics features to display
 ### Fixed
 
 - [#267](https://github.com/jamwaffles/embedded-graphics/pull/267) Fonts with CHARACTER_SPACING now properly render the background color between characters.
+- Fixed Chars like '°' and similar for Font12x16 & Font24x32
```
```suggestion
- [#276](https://github.com/jamwaffles/embedded-graphics/pull/276) Fixed Chars like '°' and similar for `Font12x16` & `Font24x32`
```

caemor commented:
Added 2 specific testcases (similar to the 6x8 one) and updated/improved simulator example even though I am unsure if it is really useful.

Build.sh was running succesfully until not finding the linkchecker.
